features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Subscriptions/SubscriptionPoliciesManage.jsx

```diff
@@ -0,0 +1,173 @@
+        paddingBottom: theme.spacing.unit * 2,
+    },
+    grid: {
+        margin: theme.spacing.unit * 1.25,
+    },
+    gridLabel: {
+        marginTop: theme.spacing.unit * 1.5,
+    },
+});
+
+/**
+ * Manage subscription policies of the API
+ * */
+class SubscriptionPoliciesManage extends Component {
+    constructor(props) {
+        super(props);
+        this.api = props.api;
+        this.state = {
+            subscriptionPolicies: {},
+            selectedSubscriptionPolicies: props.api.policies,
+            updateInProgress: false,
+        };
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    componentDidMount() {
+        API.policies('subscription')
+            .then((res) => {
+                this.setState({ subscriptionPolicies: res.obj.list });
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.error(error);
+                }
+            });
+    }
+
+    /**
+     * Handle onChange of selected subsription policies
+     *
+     * @param event onChange event
+     */
+    handleChange(event) {
+        this.setState({ updateInProgress: true });
+
+        const apiClient = new API();
+        const { name, checked } = event.target;
+        const { selectedSubscriptionPolicies } = this.state;
+        const { intl } = this.props;
+        let updatedSelectedPolicies = [...selectedSubscriptionPolicies];
+        const updatedAPI = { ...this.api };
+
+        if (checked) {
+            updatedSelectedPolicies[updatedSelectedPolicies.length] = name;
+        } else {
+            updatedSelectedPolicies = updatedSelectedPolicies.filter(policy => policy !== name);
+        }
+        updatedAPI.policies = updatedSelectedPolicies;
+        apiClient.update(updatedAPI)
+            .then((res) => {
+                // todo remove the following es-lint disable when update API method call return is changed
+                // eslint-disable-next-line no-underscore-dangle
+                this.api = res._data;
+                this.setState({ selectedSubscriptionPolicies: updatedSelectedPolicies, updateInProgress: false });
+                Alert.info(intl.formatMessage({
+                    id: 'Apis.Details.Subscriptions.SubscriptionPoliciesManage.policy.update.success',
+                    defaultMessage: 'API subscription policies updated successfully',
+                }));
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.error(error);
+                }
+                this.setState({ updateInProgress: false });
+                Alert.error(intl.formatMessage({
+                    id: 'Apis.Details.Subscriptions.SubscriptionPoliciesManage.policy.update.error',
+                    defaultMessage: 'Error occurred while updating subscription policies',
+                }));
+            });
```

Comment:
We could use `.finally()` to stop the loading animation `this.setState({ updateInProgress: false });` (and/or any other cleanup tasks)

Reply:
Fixed